### PR TITLE
Removing extra cases in GetPropertyValue from from children classes.

### LINKED ISSUE #6236: Move returning of the virtual properties to `AccessibleObject` from children classes
There're some virtual properties at the base `AccessibleObject` class, like `BoundingRectangle`, that are returned in the `GetPropertyValue` method of the children classes. Since these properties are virtual, we can return them in the `GetPropertyValue` method at the base class, even if they are overridden at the children classes.

_Originally posted by @dreddy-work in https://github.com/dotnet/winforms/pull/5745#discussion_r740377554_

### PR DESCRIPTION
**Pull request by NikitaSemenovAkvelon**
Fixes #6236

## Proposed changes

- Moved returning values of the virtual properties to the base class.
- Ordered switch cases in `AccessibleObject.GetPropertyValue` alphabetically.
- Removed extra `RecnatgleBounds` overridings.
- Used switch expression instead of if blocks.

## Customer Impact

- It's a technical refactoring.

## Regression? 

- No

## Risk

- Minimal

## Test methodology 

- Unit tests

 

## Test environment(s) 

- 7.0.100-preview.5.22307.18

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7538)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
Adding `no-merge` label to take this in NET8 because this change is a refactoring and thus does not meet the NET7 bug bar.

**Comment by Tanya-Solyanik** [comment]
Please send privates to CTIs to test this change, but let them know that this is a low priority task because it will go into NET8.

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ComboBoxChildDropDownButtonUiaProvider.cs [hunk 1]
```diff
@@ -119,8 +119,6 @@ internal override int GetChildId()
             {
                 switch (propertyID)
                 {
-                    case UiaCore.UIA.BoundingRectanglePropertyId:
-                        return BoundingRectangle;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.ComboBoxChildDropDownButtonUiaProvider.cs:123]: nit: use switch expressions in all `GetPropertyValue` overrides

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObject.cs [hunk 2]
```diff
@@ -21,18 +21,16 @@ public ToolStripDropDownButtonAccessibleObject(ToolStripDropDownButton ownerItem
...
-                    return UiaCore.UIA.ButtonControlTypeId;
-                }
-
-                return base.GetPropertyValue(propertyID);
-            }
+                    // ToolStripDropDownItemAccessibleObject implements a default Role as MenuItem
+                    // because of this, ToolStripItemAccessibleObject will return the unexpected result for this.
+                    // Return Button as the expected value by default
+                    UiaCore.UIA.ControlTypePropertyId when _owningToolStripDropDownButton.AccessibleRole ==
+                                                           AccessibleRole.Default => UiaCore.UIA.ButtonControlTypeId,
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDownButton.ToolStripDropDownButtonAccessibleObject.cs:31]: I would format like this
```
UiaCore.UIA.ControlTypePropertyId when 
    _owningToolStripDropDownButton.AccessibleRole == AccessibleRole.Default 
    => UiaCore.UIA.ButtonControlTypeId,
```

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks very good, I added some nits.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Comment by Olina-Zhang** [comment]
Tested this PR with changed controls in Accessibility tools, no new issue was found.

**Comment by NikitaSemenovAkvelon** [comment]
CTI approved.

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**